graph/src/components/store.rs

```diff
@@ -194,19 +210,39 @@ impl EntityChange {
 /// any specific blocks, `StoreEvents` for it are generated as soon as they are
 /// written to the store.
 pub struct StoreEvent {
+    /// The event source (block number/hash) of the event. Mostly here to help
+    /// with tracing how events go through the system in logs.
     pub source: EventSource,
     pub changes: Vec<EntityChange>,
 }
 
-/// A boxed `StoreEventStream`
-pub type StoreEventStreamBox = Box<Stream<Item = StoreEvent, Error = ()> + Send>;
+impl StoreEvent {
+    /// Extend `ev1` with `ev2`. If `ev1` is `None`, just set it to `ev2`
+    fn accumulate(ev1: &mut Option<StoreEvent>, ev2: StoreEvent) {
+        if let Some(e) = ev1 {
+            e.extend(ev2);
+        } else {
+            ev1.replace(ev2);
+        }
+    }
+
+    fn extend(&mut self, other: StoreEvent) {
+        if other.source > self.source {
```

Comment:
Considering this is just for logging, can we not have this comparison and just set `self.source = other.source` unconditionally? Then we can avoid having to define `PartialOrd` at all for block pointers.

Reply:
As discussed in Discord, I ripped out the `PartialOrd` stuff for `EventSource`, and removed `source` from `StoreEvent`; it now has a `tag: usize` to help distinguish events in the logs to make it possible to trace where events that were sent are received.